interfaces/builtin/steam_support.go

```diff
@@ -0,0 +1,160 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const steamSupportSummary = `allow Steam to configure pressure-vessel containers`
+
+const steamSupportBaseDeclarationPlugs = `
+  steam-support:
+    allow-installation: false
+    deny-auto-connection: true
+`
+
+const steamSupportBaseDeclarationSlots = `
+  steam-support:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const steamSupportConnectedPlugAppArmor = `
+# Allow pressure-vessel to set up its Bubblewrap sandbox.
+/sys/kernel/ r,
+@{PROC}/sys/kernel/overflowuid r,
+@{PROC}/sys/kernel/overflowgid r,
+@{PROC}/sys/kernel/sched_autogroup_enabled r,
+@{PROC}/pressure/io r,
+@{PROC}/@{pid}/uid_map rw,
+@{PROC}/@{pid}/gid_map rw,
+@{PROC}/@{pid}/setgroups rw,
+@{PROC}/@{pid}/mounts r,
+@{PROC}/@{pid}/mountinfo r,
```

Comment:
Can these `@{PROC}/@{pid}/` entries be reduced with an `owner` prefix? Unfortunately `@{pid}` doesn't yet mean "only this process's pid", so these rules are perhaps wider than intended.